### PR TITLE
[test] Add tests which convert from issues

### PR DESCRIPTION
**Pull request by wanghongjuan**
- Add tests for issues: #6143 #6113 #6001 #5980 #5943 #5882 #6231
  #6229 #6171 #6099, totally 10 tests.

- Tested on linux, Windows 10 and MacOS, all of them are passed.

### CODE REVIEW

File: test/sanity/issue6001-devtools-open-crash/test.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+except:
+  pass
+
+assert(False == os.path.isfile(binfile))
+subprocess.call([nwjc, "--future", "index.js", "index.bin"])
+assert(os.path.isfile(binfile))
+
+driver = webdriver.Chrome(executable_path=os.environ['CHROMEDRIVER'], chrome_options=chrome_options)
+driver.implicitly_wait(2)
+time.sleep(20)
```
Comment (rogerwang) [inline test/sanity/issue6001-devtools-open-crash/test.py]: Please remove such sleep calls. It will slow down running tests.

Comment (wanghongjuan) [reply test/sanity/issue6001-devtools-open-crash/test.py]: Removed all unnecessary sleep calls. PTAL

File: test/sanity/issue5882-debugging-tools-crash/test.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    wait_window_handles(driver, 2)
+    print 'switch to devtools'
+    switch_to_devtools(driver)
+    print "click Sources panel"
+    time.sleep(1)
+    devtools_click_tab(driver, 'sources')
+    print 'start to debug line 16 via breakpoint'
+    for i in range(10):
+        print "click Resume script execution button: %s" % i 
+        time.sleep(5)
```
Comment (rogerwang) [inline test/sanity/issue5882-debugging-tools-crash/test.py]: Here is the example of more sleep calls. Please remove them or replace it with a loop like this: https://github.com/nwjs/nw.js/blob/nw27/test/sanity/nw_util.py#L103

Comment (wanghongjuan) [reply test/sanity/issue5882-debugging-tools-crash/test.py]: Replaced the sleep call following the example that your mentioned. But for waiting for crash tests,  this method seems not a good choose,  do you have any better solution?

Comment (rogerwang) [reply test/sanity/issue5882-debugging-tools-crash/test.py]: which test do you mean?

File: test/sanity/issue6001-devtools-open-crash/test.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+driver.implicitly_wait(2)
+try:
+    switch_to_app(driver)
+    print driver.current_url
+    print 'waiting for devtools open'
+    wait_window_handles(driver, 2)
+    print 'switch to devtools'
+    switch_to_devtools(driver, None, True)
+    print 'waiting for crash'
+    time.sleep(5)
```
Comment (wanghongjuan) [inline test/sanity/issue6001-devtools-open-crash/test.py]: @rogerwang just like this, I need to set the time waiting for crash.

Comment (rogerwang) [reply test/sanity/issue6001-devtools-open-crash/test.py]: this case is special and it is ok to sleep 5 seconds to make sure there is no crash in this period.

Comment (wanghongjuan) [reply test/sanity/issue6001-devtools-open-crash/test.py]: OK, if there is no other question, could you please merge this PR? Thanks a lot.

**Comment by rogerwang** [comment]
Could you please verify for each case, it will fail with the NW.js version before it was fixed? This would help to verify the case is effective. The version in which bugs were fixed can be found in the CHANGELOG.

**Comment by wanghongjuan** [comment]
OK, I will verify them again with latest release SDK on MacOS, Ubuntu and Windows.

**Comment by rogerwang** [comment]
verifying on one platform should be ok.

**Comment by wanghongjuan** [comment]
All tests are verified with the failed NW.js version, except #5882 and #5943.

File: test/sanity/issue6143-mac-setProgressBar-crash/index.html [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+<html>
+<head>
+  <meta charset="utf-8">
+  <meta http-equiv="X-UA-Compatible" content="IE=edge">
+  <title>issue6143-mac-setProgressBar-crash</title>
+</head>
+<body>
+  <h3 id="result">success</h3>
+  <script>
+    chrome.downloads.download({ url: "http://releases.ubuntu.com/xenial/ubuntu-16.04.3-desktop-amd64.iso" });
```
Comment (rogerwang) [inline test/sanity/issue6143-mac-setProgressBar-crash/index.html]: I think this could lead to full disk on the test machine?

**Comment by rogerwang** [comment]
btw, you could consider to submit multiple PRs to nw28 for each of the test. So the good ones can be merged first.

File: test/sanity/issue6143-mac-setProgressBar-crash/index.html (hunk 4 above)
Comment (wanghongjuan) [reply test/sanity/issue6143-mac-setProgressBar-crash/index.html]: Updated the link to https://github.com/nwjs/nw.js/archive/nw-v0.27.4.tar.gz

**Comment by wanghongjuan** [comment]
Thanks for your advice, I will submit the test one by one from next PR.

File: test/sanity/issue6143-mac-setProgressBar-crash/index.html (hunk 4 above)
Comment (rogerwang) [reply test/sanity/issue6143-mac-setProgressBar-crash/index.html]: given enough time the disk should be still full. The correct way is to find where will the download be and remove them after the test is done.

File: test/sanity/issue4056-mac-close-crash/test.py [hunk 5]
```diff
@@ -3,9 +3,9 @@
 import platform
 import sys
 
-if platform.system() != 'Darwin':
-    print 'Skipped for non Mac platform'
-    sys.exit(0)
+#if platform.system() != 'Darwin':
+#    print 'Skipped for non Mac platform'
+#    sys.exit(0)
```
Comment (rogerwang) [inline test/sanity/issue4056-mac-close-crash/test.py]: Why did you modify this case?

Comment (wanghongjuan) [reply test/sanity/issue4056-mac-close-crash/test.py]: Oops, that's a mistake, recovered it.

**Comment by wanghongjuan** [comment]
Submitted new PRs for these issues, close this one

**Pull request closed without merging**